storage/src/device.rs

```diff
@@ -669,10 +671,6 @@ impl BlobIoSegment {
 
     #[inline]
     pub fn append(&mut self, _offset: u32, len: u32) {
-        debug_assert!(self.offset + self.len == _offset);
-        debug_assert!(_offset.checked_add(len).is_some());
```

Comment:
Should we still keep these two checks?

Reply:
Let's keep it. I have to preference for it